pkg/registry/apis/dashboard/sub_dto.go

```diff
@@ -128,33 +127,24 @@ func (r *DTOConnector) Connect(ctx context.Context, name string, opts runtime.Ob
 			return
 		}
 
-		// Calculate access information -- needed to help smooth transition from /api/dashboard format
-		dto := &dashboards.Dashboard{
-			UID:   name,
-			OrgID: info.OrgID,
-			ID:    obj.GetDeprecatedInternalID(), // nolint:staticcheck
-		}
-		manager, ok := obj.GetManagerProperties()
-		if ok && manager.Kind == utils.ManagerKindPlugin {
-			dto.PluginID = manager.Identity
-		}
-
-		guardian, err := guardian.NewByDashboard(ctx, dto, info.OrgID, user)
-		if err != nil {
-			responder.Error(err)
-			return
-		}
-		canView, err := guardian.CanView()
+		dashScope := dashboards.ScopeDashboardsProvider.GetResourceScopeUID(name)
+		evaluator := accesscontrol.EvalPermission(dashboards.ActionDashboardsRead, dashScope)
+		canView, err := r.accessControl.Evaluate(ctx, user, evaluator)
 		if err != nil || !canView {
 			responder.Error(fmt.Errorf("not allowed to view"))
 			return
 		}
 
 		access := &dashboard.DashboardAccess{}
-		access.CanEdit, _ = guardian.CanEdit()
-		access.CanSave, _ = guardian.CanSave()
-		access.CanAdmin, _ = guardian.CanAdmin()
-		access.CanDelete, _ = guardian.CanDelete()
+		writeEvaluator := accesscontrol.EvalPermission(dashboards.ActionDashboardsWrite, dashScope)
+		access.CanSave, _ = r.accessControl.Evaluate(ctx, user, writeEvaluator)
+		access.CanEdit = access.CanSave
+		adminEvaluator := accesscontrol.EvalAll(
+			accesscontrol.EvalPermission(dashboards.ActionDashboardsPermissionsRead, dashScope),
+			accesscontrol.EvalPermission(dashboards.ActionDashboardsPermissionsWrite, dashScope))
+		access.CanAdmin, _ = r.accessControl.Evaluate(ctx, user, adminEvaluator)
+		deleteEvaluator := accesscontrol.EvalPermission(dashboards.ActionDashboardsDelete, dashScope)
+		access.CanDelete, _ = r.accessControl.Evaluate(ctx, user, deleteEvaluator)
```

Comment:
We repeat this quite a lot, create some sort of helper function(s) for this. Ideally we would use a common way to set this kind of data but it could make sense as long we can `can*` properties

Reply:
I don't agree nor disagree, the way it is right now is at least explicit and easy to understand what's going on which might help people onboard to the code (and we have a bunch of them right now)

Reply:
Yeah sure, most it it is straight forward, canSave uses write, canDelete uses delete etc.

I guess this is the only "problematic" part 
```
adminEvaluator := accesscontrol.EvalAll(
			accesscontrol.EvalPermission(dashboards.ActionDashboardsPermissionsRead, dashScope),
			accesscontrol.EvalPermission(dashboards.ActionDashboardsPermissionsWrite, dashScope))
```

Reply:
easy, we drop the `ActionDashboardsPermissionsRead` :innocent:

Reply:
this